api/stream/design.go

```diff
@@ -39,7 +42,21 @@ const (
 
 // nolint: gochecknoinits
 func init() {
-	dependencies.RegisterPlugin("github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies")
+	dependencies.RegisterPlugin(dependencies.Config{
+		Package: "github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies",
+		DependenciesTypeFn: func(method *service.MethodData) string {
+			if dependencies.HasSecurityScheme("APIKey", method) {
+				return "dependencies.ProjectRequestScope"
+			}
+			return "dependencies.PublicRequestScope"
+		},
+		DependenciesProviderFn: func(method *service.EndpointMethodData) string {
+			if dependencies.HasSecurityScheme("APIKey", method.MethodData) {
+				return "ctx.Value(dependencies.ProjectRequestScopeCtxKey).(dependencies.ProjectRequestScope)"
+			}
+			return "ctx.Value(dependencies.PublicRequestScopeCtxKey).(dependencies.PublicRequestScope)"
+		},
+	})
```

Comment:
- `package` attribute to `Config.Package`.
- Added 2 new functions, they replaces hard-coded templates bellow.